azure/storage_account/variables.tf

```
@@ -18,37 +18,19 @@ variable "name" {
   }
 }
 
-variable "location" {
-  # For a complete list of available Azure regions run at cli:  
-  # az account list-locations  --query "[].{displayName:displayName, location:name}" --output table
-  description = "(Optional) The Azure Region where the resource should exist."
-  type        = string
-  default     = "westeurope"
-
-  validation {
-    condition     = contains(["westeurope", "northeurope"], var.location)
-    error_message = "The 'location' value is invalid. Valid options are 'westeurope', 'northeurope'."
-  }
-}
-
 variable "resource_group_name" {
-  description = "The name of an existing Resource Group."
   type        = string
-
-  validation {
-    condition     = can(coalesce(var.resource_group_name))
-    error_message = "The 'resource_group_name' value is invalid. It must be a non-empty string."
-  }
+  description = "The name of an existing Resource Group."
 }
```

Please, consider keeping the `resource_group_name` validation ensuring that the variable is a non-empty string as we do in the `name` variable
```suggestion
variable "resource_group_name" {
  description = "The name of an existing Resource Group."
  type        = string

  validation {
    condition     = can(coalesce(var.resource_group_name))
    error_message = "The 'resource_group_name' value is invalid. It must be a non-empty string."
  }
}
```

---

But this is not necessary since it will always check the name given from the RG. From the data.

---

agree...